### PR TITLE
Implement BlobsSidecarsByRange validation

### LINKED ISSUE #6584: Implement BlobsSidecarsByRange validation
Perform the validation required: https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/p2p-interface.md#blobssidecarsbyrange-v1

### PR DESCRIPTION
**Pull request by StefanBratanov**
## Fixed Issue(s)
fixes #6584 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobsSidecarsByRangeMessageHandler.java [hunk 1]
```diff
@@ -116,6 +142,22 @@ protected void onIncomingMessage(
...
+   * v1</a>
+   */
+  private boolean verifyBlobsSidecarsAreAvailable(
+      final Optional<UInt64> earliestAvailableSidecarEpoch, final UInt64 currentEpoch) {
+    if (earliestAvailableSidecarEpoch.isEmpty()) {
+      return false;
+    }
+    final UInt64 startEpoch =
+        eip4844ForkEpoch.max(
+            currentEpoch.safeMinus(MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS).orElse(ZERO));
```
Comment (tbenr) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobsSidecarsByRangeMessageHandler.java]: `currentEpoch.minusMinZero`?

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobsSidecarsByRangeMessageHandler.java [hunk 2]
```diff
@@ -69,6 +77,13 @@ public BlobsSidecarsByRangeMessageHandler(
   @Override
   public Optional<RpcException> validateRequest(
       final String protocolId, final BlobsSidecarsByRangeRequestMessage request) {
+    if (request.getCount().isGreaterThan(maxRequestSize)) {
```
Comment (tbenr) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobsSidecarsByRangeMessageHandler.java]: Not sure about this. We need to make sure we do not respond with more than `maxRequestSize` (yeah the name is misleading) but we can accept a bigger range, like what we do in `BeaconBlocksByRangeMessageHandler`. Reading the specs seems to be aligned with the blocks by range behaviour.

Comment (StefanBratanov) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobsSidecarsByRangeMessageHandler.java]: agree, removed this validation

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**